lib/fileset/tests.sh

```diff
@@ -57,18 +57,35 @@ with lib.fileset;'
 expectEqual() {
     local actualExpr=$1
     local expectedExpr=$2
-    if ! actualResult=$(nix-instantiate --eval --strict --show-trace \
+    if actualResult=$(nix-instantiate --eval --strict --show-trace 2>"$tmp"/actualStderr \
         --expr "$prefixExpression ($actualExpr)"); then
-        die "$actualExpr failed to evaluate, but it was expected to succeed"
+        actualExitCode=$?
```

Comment:
Bash hack because we have `set -e` here, meaning if we just do
```bash
some-potentially-failing-command
actualExitCode=$?
```

Then it would exit when the command fails. And if we do

```bash
some-potentially-failing-command || true
actualExitCode=$?
```

Then `actualExitCode` would always be 0

We basically have two options, either the `if` as here:
```bash
if some-potentially-failing-command; then
  # Or setting it to 0 would be the same
  actualExitCode=$?
else
  actualExitCode=$?
fi
```

Or to temporarily unset `set -e`:
```bash
set +e
some-potentially-failing-command
actualExitCode=$?
set -e
```

Both kind of suck, I don't have a preference :)